src/main/scala/controllers/ActionHandler.scala

```diff
@@ -102,27 +102,22 @@ class ActionHandler @Inject()(applications: ApplicationOps, applicationForms: Ap
   def redirectToPreview(id: ApplicationId, sectionNumber: Int) =
     Redirect(routes.ApplicationPreviewController.previewSection(id, sectionNumber))
 
-  def renderSectionForm(id: ApplicationId,
+  def renderSectionForm(app: ApplicationDetail,
```

Comment:
What I meant is that if something returns just Result, but not Future[Result], then it can be called synchronously from the controller. Future[Result] return type on the public methods prevents that.

That's a minor issue, though.